golang/channel.go

```diff
@@ -35,6 +35,8 @@ import (
 var (
 	errAlreadyListening  = errors.New("channel already listening")
 	errInvalidStateForOp = errors.New("channel is in an invalid state for that operation")
+
+	ErrNoServiceName = errors.New("No service name provided")
```

Comment:
Can you also add a comment since the variable is exported:
```go
// ErrNoServiceName is returned when creating a channel with no service name.
// A service name is required for [...]
```
Also, make error lowercase ("no service name provided") - all go errors are supposed to be lower case.